backend/app/api/daily_journal.py

```
@@ -0,0 +1,286 @@
+    """
+    try:
+        dj = await daily_journal_store.generate(
+            db=db,
+            user_id=current_user.id,
+            work_date=body.work_date,
+            entry_ids=body.entry_ids,
+            overwrite=body.overwrite,
+            farm_name=current_user.farmname or current_user.name,
+        )
+    except DailyJournalError as e:
+        raise HTTPException(status_code=e.status, detail={"code": e.code, "message": e.message})
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)